tests/Feature/UserProfileTest.php

```diff
@@ -4,124 +4,145 @@
     {
-        $response = $this->getJson("/api/profile");
+        $this->actingAs($this->user)
+            ->putJson("/api/profile", [
+                "first_name" => "Updated",
+                "last_name" => "Name",
+            ])
+            ->assertOk()
+            ->assertJsonPath("data.first_name", "Updated")
+            ->assertJsonPath("data.last_name", "Name");
+    }
 
-        $response->assertUnauthorized();
+    public function testUpdateProfileRequiresFirstName(): void
+    {
+        $this->actingAs($this->user)
+            ->putJson("/api/profile", [
+                "last_name" => "OnlyLastName",
+            ])
+            ->assertJsonValidationErrors("first_name");
     }
 
-    public function testUserCanUpdateNameAndLastName(): void
+    public function testGuestCannotUpdateProfile(): void
     {
-        $user = User::factory()->create();
+        $this->putJson("/api/profile", [
+            "first_name" => "Nope",
+            "last_name" => "StillNope",
+        ])->assertUnauthorized();
+    }
 
-        Sanctum::actingAs($user);
+    public function testProfileUpdateDoesNotAcceptTooLongNames(): void
+    {
+        $longName = str_repeat("a", 300);
+
+        $this->actingAs($this->user)
+            ->putJson("/api/profile", [
+                "first_name" => $longName,
+                "last_name" => "Lastname",
+            ])
+            ->assertJsonValidationErrors("first_name");
+    }
 
-        $payload = [
-            "first_name" => "UpdatedName",
-            "last_name" => "UpdatedSurname",
-        ];
+    public function testUserGetsSameResponseWhenAccessingOwnProfileById(): void
+    {
+        $this->actingAs($this->user);
 
-        $response = $this->putJson("/api/profile", $payload);
+        $response = $this->getJson("/api/profile/{$this->user->id}");
 
-        $response->assertOk()
+        $response->assertStatus(302)
             ->assertJson([
-                "message" => "Profile updated successfully.",
-                "data" => [
-                    "first_name" => "UpdatedName",
-                    "last_name" => "UpdatedSurname",
-                    "email" => $user->email,
-                ],
+                "message" => "profile.redirected",
+                "redirect" => "/api/profile",
             ]);
-
-        $this->assertDatabaseHas("users", [
-            "id" => $user->id,
-            "first_name" => "UpdatedName",
-            "last_name" => "UpdatedSurname",
-        ]);
     }
 
-    public function testUpdateProfileRequiresFirstName(): void
+    public function testAuthenticatedUserCanViewOtherUsersProfile(): void
     {
-        $user = User::factory()->create();
-
-        Sanctum::actingAs($user);
-
-        $response = $this->putJson("/api/profile", [
-            "last_name" => "OnlyLastName",
-        ]);
-
-        $response->assertStatus(422);
-        $response->assertJsonValidationErrors(["first_name"]);
+        $this->actingAs($this->user)
```

Comment:
If we use `$otherUser` only in this test, we don't need to create it setUp method